modules/elasticsearch/data-nodes.tf

```diff
@@ -191,12 +239,12 @@ resource "aws_alb_target_group" "elasticsearch-api-secured" {
 }
 
 resource "aws_alb_listener" "elasticsearch-api-secured" {
-  count             = "${var.external_alb_setup ? 1 : 0}"
+  count             = "${var.external_alb_setup && !lookup(var.external_alb, "deploy_elb", false) ? 1 : 0}"
   load_balancer_arn = "${lookup(var.external_alb, "arn", "")}"
   port              = 9201
   protocol          = "HTTPS"
   ssl_policy        = "ELBSecurityPolicy-2016-08"
-  certificate_arn   = "${data.aws_acm_certificate.elasticsearch-cert.arn}"
+  certificate_arn   = "${var.external_alb["certificate"]}"
```

Comment:
I would consider renaming the key in `external_alb` map to `certificate_arn`.

Reply:
I was contemplating the same thing. Changed.